src/transformers/generation/logits_process.py

```diff
@@ -1102,3 +1102,82 @@ def __call__(self, input_ids, scores):
+
+    Args:
+        guidance_scale (float):
+            The guidance scale for classifier free guidance (CFG). CFG is enabled by setting `guidance_scale > 1`.
+            Higher guidance scale encourages the model to generate samples that are more closely linked to the input
+            prompt, usually at the expense of poorer quality.
+        unconditional_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`):
+            Indices of input sequence tokens in the vocabulary for the unconditional branch.
+        model:
+            The LM computing the unconditional scores. Supposedly the same as the one computing the conditional scores.
+            Both models must use the same tokenizer.
+        smooth_factor (float):
+            The interpolation weight for CFG Rescale. 1 means no rescaling, 0 reduces to the conditional scores without
+            CFG. Turn it lower if the output degenerates.
+
+    Examples:
+
+    ```python
+    >>> # base prompt emphasis example
+    >>> from transformers import AutoTokenizer, AutoModelForCausalLM
+
+    >>> model = AutoModelForCausalLM.from_pretrained("gpt2")
+    >>> tokenizer = AutoTokenizer.from_pretrained("gpt2")
+    >>> inputs = tokenizer(["Today, a dragon flew over Paris, France,"], return_tensors="pt")
+    >>> summary_ids = model.generate(inputs["input_ids"], guidance_scale=1.5)
+    >>> print(tokenizer.batch_decode(summary_ids, skip_special_tokens=True)[0])
+    The dragon flew over Paris, France, landing in Lyon, a city of a few million. Dragon-flying was a new form of
+    transport, and the dragon was the first in Europe.
+
+    >>> # with a negative prompt
+    >>> neg_inputs = tokenizer(["A very happy event happened,"], return_tensors="pt")
+    >>> summary_ids = model.generate(
+    ...     inputs["input_ids"], guidance_scale=5, cfg_rescale=0.3, negative_prompt=neg_inputs["input_ids"]
+    ... )
+    >>> print(tokenizer.batch_decode(summary_ids, skip_special_tokens=True)[0])
+    The dragon flew over Paris, France, crashing into Notre Dame Cathedral in the French capital killing at least 127
+    people and injuring more than 350.
+    ```
+    """
+
+    def __init__(self, guidance_scale, unconditional_ids, model, rescale_factor=1.0):
+        self.guidance_scale = guidance_scale
+        self.unconditional_ids = unconditional_ids
+        self.model = model
+        self.out = None
+        self.rescale_factor = rescale_factor
+
+    def __call__(self, input_ids, scores):
+        scores = torch.nn.functional.log_softmax(scores, dim=-1)
+        if self.guidance_scale == 1:
+            return scores
+
+        if self.out is None:
+            self.out = self.model(
+                self.unconditional_ids, attention_mask=torch.ones_like(self.unconditional_ids[:, -1:]), use_cache=True
+            )
+        else:
+            self.out = self.model(
+                input_ids[:, -1:],
+                attention_mask=torch.ones_like(input_ids[:, -1:]),
```

Comment:
Using a Llama model this gives:
```
ValueError: Attention mask should be of size (1, 1, 1, 7), but is torch.Size([1, 1, 1, 1])
```
The error is coming from `transformers/models/llama/modeling_llama.py#222` if it helps figure out how you should adjust it.  But again, I'm not sure what the value of setting an attention mask on a single token is.  If I remove the `attention_mask` the code seems to work.

Unrelatedly, I think you should do the `guidance_scale == 1` check before you do the `log_softmax` just to not perturb the logits if you're not going to do anything with them. It probably doesn't matter, but it's one less set of 32,000 operations, which is always nice.

Reply:
@cyberfox can you confirm whether your model has `model.config.use_cache` set to `True`? Some LLaMA checkpoints have it set to `False`, and it would explain the issue you've seen.

If you're still seeing an issue after checking my suggestion above -- let's wait for this PR to be merged, and then open a separate issue :) That way we can focus on shipping the common usage case, and fix corner cases afterward!